x-pack/legacy/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -152,7 +152,7 @@ export class TaskRunner {
       params,
       state: alertTypeState,
       startedAt: this.taskInstance.startedAt!,
-      previousStartedAt,
+      previousStartedAt: previousStartedAt && new Date(previousStartedAt),
```

Comment:
Looking at this code in vsCode is confusing.  The previous reference to `previousStartedAt` in this method, above, is typed as `any`, since `this.taskInstance.state` is typed to `Record<string, any>`.  But then in this call, before the new date conversion, it's showing that it's typed as `Date | undefined`.  How did it make the leap there?

Reply:
It's because in Alerting we pass the value of `startedAt` in as a variable called `previousStartedAt` (a field on TaskInstance) which is a `Date | undefined`.
We then store it in the state.

When pulled out of the state, like all state, it's an `any` and needs to be explicitly cast back to a JS Date.

This would be easier if TaskInstance were a generic which took the shape of the state, but you'd still need to cast it somewhere.

Reply:
I never realized you could assign an `any` typed value to any type of variable - `any` goes both ways!  How did I not know that?

Seems like we don't have a great way of dealing with ES props / SO attributes <-> TS types, we seem to do it by hand a lot - feels like we should do some kind of validation here, not required for this PR though.